planner/core/exhaust_physical_plans.go

```diff
@@ -2332,13 +2334,13 @@ func (la *LogicalAggregation) getHashAggs(prop *property.PhysicalProperty) []Phy
 	if la.ctx.GetSessionVars().AllowBCJ {
 		taskTypes = append(taskTypes, property.CopTiFlashLocalReadTaskType)
 	}
-	if la.ctx.GetSessionVars().AllowMPPExecution && la.checkCanPushDownToMPP() {
+	canPushDownToMPP := la.ctx.GetSessionVars().AllowMPPExecution && la.checkCanPushDownToMPP()
+	if canPushDownToMPP {
 		taskTypes = append(taskTypes, property.MppTaskType)
 	}
 	if la.HasDistinct() {
-		// TODO: remove AllowDistinctAggPushDown after the cost estimation of distinct pushdown is implemented.
-		// If AllowDistinctAggPushDown is set to true, we should not consider RootTask.
-		if !la.ctx.GetSessionVars().AllowDistinctAggPushDown {
+		// TODO: remove after the cost estimation of distinct pushdown is implemented.
+		if !la.ctx.GetSessionVars().AllowDistinctAggPushDown && !canPushDownToMPP {
```

Comment:
this logic is wrong.
If AllowDistinctAggPushDown is set to false, we should only consider RootTask.

Reply:
also need a test to ensure this.

Reply:
IIRC @hanfei1991 stands that we don't share the same config with `AllowDistinctAggPushDown` which is introduced by https://github.com/pingcap/tidb/pull/15500 and focuses on rewrite scalar `distinct` into `group by`.

Reply:
Yeah, the "AllowDistinctAggPushDown" doesn't decide whether to push mpp query.

Reply:
after disscussed with @zanmato1984 and @hanfei1991 , I approve this change, even through it induces some unexceptions.